### PR TITLE
Let SPARK_VERSION to specify a URL to download a Spark archive from.

### PR DESCRIPTION
**Pull request by ewencp**
As an alternative to a git hash or known version of Spark, detect
URLs by the presence of :// and download a custom Spark archive.

### CODE REVIEW

**Comment by ewencp** [comment]
This has a corresponding commit to the main Spark code: https://github.com/apache/incubator-spark/pull/455

**Comment by nchammas** [comment]
@shivaram @ewencp Is this PR still relevant? It was opened a while ago. (Just asking as a curious passer-by.)

File: templates/root/spark/conf/spark-env.sh [hunk 1]
```diff
@@ -17,7 +17,7 @@ export SCALA_HOME="/root/scala"
 export SPARK_MEM={{default_spark_mem}}
 
 # Set JVM options and Spark Java properties
-SPARK_JAVA_OPTS+=" -Dspark.local.dir={{spark_local_dirs}}"
+SPARK_JAVA_OPTS+=" -Dspark.worker.timeout=600 -Dspark.local.dir={{spark_local_dirs}}"
```
Comment (nchammas) [inline templates/root/spark/conf/spark-env.sh:20]: This change seems unrelated to the main message of the PR.

**Comment by JoshRosen** [comment]
This looks pretty old and the link to the corresponding Spark PR is broken.  I grepped through the Spark git log for `ewencp` and it doesn't look like the Spark half was merged.  Therefore, I'm going to close this PR for now (since it's nearly a year old).  Feel free to re-open this if you feel that it's still relevant.
